apps/analysis/tests/test_mutations.py

```diff
@@ -1326,3 +1326,176 @@ def _query_check(_id, **kwargs):
+        self.project_with_af.add_member(self.readonly_member_user, role=self.project_role_reader_non_confidential)
+        self.project_with_af.add_member(self.member_user, role=self.project_role_member)
+
+    def test_create_analysis_without_pillar(self):
+        minput = dict(
+            analysisData=dict(
+                title='Test Analysis', teamLead=self.member_user.id, endDate='2020-01-01'
+            ),
+            projectId=self.project_with_af.id,
+        )
+        # -- Without login
+        self.query_check(self.CREATE_MUTATION, variables=minput, assert_for_error=True)
+
+        # -- With login (non-member)
+        self.force_login(self.non_member_user)
+        self.query_check(self.CREATE_MUTATION, variables=minput, assert_for_error=True)
+
+        # --- member user (read-only)
+        self.force_login(self.readonly_member_user)
+        self.query_check(self.CREATE_MUTATION, variables=minput, assert_for_error=True)
+
+        # --- member user
+        self.force_login(self.member_user)
+        content = self.query_check(self.CREATE_MUTATION, variables=minput)
+
+        self.assertEqual(
+            content['data']['project']['analysisCreate']['result']['title'],
+            minput['analysisData']['title']
+        )
+        self.assertEqual(
+            content['data']['project']['analysisCreate']['result']['teamLead']['id'],
+            str(self.member_user.id)
+        )
+        self.assertEqual(
+            content['data']['project']['analysisCreate']['result']['endDate'],
+            str(minput['analysisData']['endDate'])
+        )
+
+    def test_create_analysis_without_analysis_framework(self):
+        minput = dict(
+            analysisData=dict(
+                title='Test Analysis', teamLead=self.member_user.id, endDate='2020-01-01'
+            ),
+            projectId=self.project_without_af.id,
+        )
+
+        self.force_login(self.non_member_user)
+        self.query_check(self.CREATE_MUTATION, variables=minput, assert_for_error=True)
+
+    def test_create_and_update_analysis(self):
+        minput = dict(
+            analysisData=dict(
+                title='Test Analysis', teamLead=self.member_user.id, endDate='2020-01-01'
+            ),
+            projectId=self.project_with_af.id,
+        )
+
+        self.force_login(self.member_user)
+        content = self.query_check(self.CREATE_MUTATION, variables=minput)
+        self.assertIsNotNone(content['data']['project']['analysisCreate']['result']['id'])
+
+        update_minput = dict(
+            analysisUpdate=dict(
+                title='Updated Analysis',
+                teamLead=self.member_user.id,
+                endDate='2022-01-01',
+            ),
+            analysisID=content['data']['project']['analysisCreate']['result']['id'],
+            projectId=self.project_with_af.id,
+        )
+        content = self.query_check(self.UPDATE_MUTATION, variables=update_minput)
```

Comment:
Let's do that whole different user check for each type of mutation (Create, Update, Delete)